core/src/main/scala/kafka/log/remote/RemoteIndexCache.scala

```diff
@@ -127,21 +127,26 @@ class RemoteIndexCache(maxSize: Int = 1024, remoteStorageManager: RemoteStorageM
   def getIndexEntry(remoteLogSegmentMetadata: RemoteLogSegmentMetadata): Entry = {
     def loadIndexFile[T <: CleanableIndex](fileName: String,
                                            suffix: String,
-                                           fetchRemoteIndex: RemoteLogSegmentMetadata => InputStream,
+                                           fetchRemoteIndex: RemoteLogSegmentMetadata => Option[InputStream],
                                            readIndex: File => T): T = {
       val indexFile = new File(cacheDir, fileName + suffix)
+      var inputStreamIsEmpty: Boolean = false
 
       def fetchAndCreateIndex(): T = {
         val inputStream = fetchRemoteIndex(remoteLogSegmentMetadata)
-        val tmpIndexFile = new File(cacheDir, fileName + suffix + RemoteIndexCache.TmpFileSuffix)
+        inputStreamIsEmpty = inputStream.isEmpty
```

Comment:
What's the use of this flag?

Reply:
I feel like there is a convoluted condition in which this may be required -

Let us say that there was an existing `remote-log-index-cache/abcdef.tx` file in the `cache` dir because of some reason. In this case, if the RLMM throws `RemoteResourceNotFoundException` exception, we do not delete the file at `remote-log-index-cache/abcdef.tx`. In that case, the `indexFile.exists()` check will pass, and the code will return an input stream of the older file.

Now, I understand that it is improbable for the file to already exist at `remote-log-index-cache/abcdef.tx`, and cannot think of a scenario when it would be so. It was just defensive coding on my part. If you think the check is not required, I can remove it.

Reply:
Ack. Let's keep the check.